### PR TITLE
[IMP] website: adapt alerts colors and options

### PR DESCRIPTION
**Pull request by qsm-odoo**
Extracted from https://github.com/odoo/odoo/pull/120302

Following the default theme redesign, this commit adapts the color of the alert component and reviews its options.

task-3888927

### CODE REVIEW

**Comment by qsm-odoo** [comment]
@stefanorigano Reviving this :) I rebased the branch. Before going further, is this still what we want? (maybe specs changed during snippets refactoring?). If yes, then I'll assign a dev to finish this :+1:

**Comment by anso-odoo** [comment]
Hello @qsm-odoo. I had a look and this is still a much appreciated and needed IMP. 

While checking I found a small issue with borders, which I fixed (basically a "border" property in `s_alert.scss` with no set border-color was making it inherit the currentColor and thus making it too dark). I also tweaked a bit the background behind the icon to make it a bit more sublte. Added these changes in a fixup so you can have a look. 

Thanks !

cc @stefanorigano 

Before: 
Screenshot 2024-04-18 at 11 29 35

After: 
Screenshot 2024-04-18 at 11 25 07

**Comment by qsm-odoo** [comment]
Rebased (fixed conflicts), squashed commits. I will now ask a dev to look into this (to make sure old alerts are treated the way it should, the code is ok, etc).

File: addons/website/static/src/snippets/s_alert/options.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        if (!alertIconEl) {
+            return;
+        }
+
+        // Note: this function is basically a "selectClass" combined with an
+        // "applyTo" but each option already comes with a "selectClass"
+        // targeting the main container. This is also why this does not need
+        // a _computeWidgetState, relying on the "selectClass" one which comes
+        // afterwards alphabetically (compared to "selectAlertIcon").
+        alertIconEl.classList.remove(...params.possibleValues.filter(v => !!v));
```
Comment (xO-Tx) [inline addons/website/static/src/snippets/s_alert/options.js:27]: @bvr-odoo It seems that this class removal doesn't handle all possible scenarios... If we remove only the `possibleValues` and we set a different icon from the media dialog, we will end up with an element holding more than one `fa-` class, which leads to inconsistencies :grimacing:

https://github.com/odoo/odoo/assets/25032876/750baed1-b7a6-4f53-8a60-185e6acea6a4

We simply need to remove all icons classes before setting the right one :+1:

**Review by xO-Tx — CHANGES_REQUESTED** [review]

**Comment by qsm-odoo** [comment]
@bvr-odoo There is no alert in the codebase where you need to increase the version number? Instead of increasing the version number, not possible to make a migration script? To confirm :)

@robodoo delegate=SergeBayet 
@SergeBayet

**Comment by bvr-odoo** [comment]
> @bvr-odoo There is no alert in the codebase where you need to increase the version number?

No

> Instead of increasing the version number, not possible to make a migration script? To confirm :)

I'm not sure about to change the design of existing pages.
If they wish, users can change their “alert” themselves. But I wouldn't change them automatically
IMO this seems a bit "intrusive"
But I could be wrong :)

**Review by SergeBayet — COMMENTED** [review]
Good PR :+1: 

As a reminder, the task-id should be updated in both the PR description and in the commit message. :smiley: 

LGTM but I have two questions regarding the UI:

1. When a link is inserted into an alert snippet, it lacks the "alert-link" class, which ensures the link color matches the alert theme. Is this intentional, or can a CSS rule be added to apply this class, such as:
   ```css
   a {
     font-weight: 600;
     color: var(--alert-link-color);
   }
   ```

2. Shouldn't the icon be vertically aligned as in the Bootstrap documentation? 
image

**Comment by bvr-odoo** [comment]
> 1- When a link is inserted into an alert snippet, it lacks the "alert-link" class, which ensures the link color matches the alert theme. Is this intentional, or can a CSS rule be added to apply this class, such as:

I won't do that, it will impact inner snippet dropped inside the "alert"

> 2- Shouldn't the icon be vertically aligned as in the Bootstrap documentation?

I can't reproduce it.. it's not the style of the new alert in your screenshot

File: addons/website/static/src/snippets/s_alert/options.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Changes alert's icon pictogram.
+     *
+     * @see this.selectClass for parameters
+     */
+    selectAlertIcon(previewMode, widgetValue, params) {
+        const alertIconEl = this.$target[0].querySelector('.s_alert_icon');
```
Comment (SergeBayet) [inline addons/website/static/src/snippets/s_alert/options.js:18]: double quotes here :)

File: addons/website/static/src/snippets/s_alert/options.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/** @odoo-module **/
```
Comment (SergeBayet) [inline addons/website/static/src/snippets/s_alert/options.js:1]: Not mandatory anymore

File: addons/website/static/src/snippets/s_alert/options.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import options from "@web_editor/js/editor/snippets.options";
+import fonts from '@web_editor/js/wysiwyg/fonts';
```
Comment (SergeBayet) [inline addons/website/static/src/snippets/s_alert/options.js:4]: double quotes here.

File: addons/website/static/src/snippets/s_alert/options.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+options.registry.Alert = options.Class.extend({
+
```
Comment (SergeBayet) [inline addons/website/static/src/snippets/s_alert/options.js:7]: My linter tells me to remove this blank line...

File: addons/website/static/src/snippets/s_alert/options.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        // a _computeWidgetState, relying on the "selectClass" one which comes
+        // afterwards alphabetically (compared to "selectAlertIcon").
+        fonts.computeFonts();
+        let iconClass = widgetValue;
+        const allFaIcons = fonts.fontIcons[0].alias;
+        if (previewMode === "reset") {
+            iconClass = this.initialIconClass;
+        } else {
+            this.initialIconClass = [...alertIconEl.classList]
+                .filter(c => allFaIcons.includes(c));
```
Comment (SergeBayet) [inline addons/website/static/src/snippets/s_alert/options.js:35]:
```suggestion
            this.initialIconClass = [...alertIconEl.classList].filter((c) =>
                allFaIcons.includes(c)
            );
```

**Review by SergeBayet — COMMENTED** [review]
LTGTM :+1: Just some lintering remarks.

File: addons/website/static/src/snippets/s_alert/options.js (hunk 5 above)
Comment (bvr-odoo) [reply addons/website/static/src/snippets/s_alert/options.js:7]: don't trust him 😸
There is always empty lignes before and after this type of comment section.
image

File: addons/website/static/src/snippets/s_alert/options.js (hunk 6 above)
Comment (bvr-odoo) [reply addons/website/static/src/snippets/s_alert/options.js:35]: 👍🏼 I changed it a bit differently than you suggested

**Review by SergeBayet — APPROVED** [review]
Thanks for the modifications. LGTM :)

**Comment by SergeBayet** [comment]
@robodoo r+

**Pull request closed without merging**